### PR TITLE
Tickets/dm34748: Converted keywords to QueryEfd.py library functions

### PR DESCRIPTION
**Pull request by rbovill**

### CODE REVIEW

File: QueryEfd.py [hunk 1]
```diff
@@ -264,8 +526,47 @@ def verify_topic_attribute(self, csc: str, topic: str, fields: list, expected_va
...
+        topic_2 : `str`
+            The name of the second topic.
+        time_window : `int`
+            The value, in seconds, under which publish times
+            should differ..
+        index : `int`
+            The index of the CSC, if applicable (default is None).
+        """
+        timestamp1 = self.get_topic_sent_time(csc, topic_1)
+        time_1 = time.mktime(
```
Comment (mareuter) [inline QueryEfd.py:553]: Not sure why the extra conversion is necessary. The `strptime` returns a `datetime` and the subtraction of the two is a `timedelta`. What was the issue that prevented this mechanism from being used?

**Review by mareuter — COMMENTED** [review]
I'd like to understand some of the choices made in the code.

File: Verify_AT_Stop.robot [hunk 2]
```diff
@@ -1,11 +1,10 @@
 *** Settings ***
-Library    QueryEfd
+Resource    Global_Vars.resource
```
[the remark this answers is not in the document]
Comment (rbovill) [reply Verify_AT_Stop.robot:2]: Fixed.

File: QueryEfd.py [hunk 3]
```diff
@@ -140,6 +202,20 @@ def get_recent_samples(
...
+    def verify_version(self, version: str) -> None:
+        """Fails if the version does not conform to SemVer syntax.
+        The version_regex is defined as a Class attribute.
+
+        Parameters
+        ----------
+        version : `str`
+            The version string.
+        """
+        pattern = re.compile(self.version_regex)
```
[the remark this answers is not in the document]
Comment (rbovill) [reply QueryEfd.py:215]: Good suggestion.  I moved it.

File: QueryEfd.py [hunk 4]
```diff
@@ -205,41 +281,227 @@ def verify_summary_state(
...
+            csc, "logevent_summaryState", fields, 4, index
         )
-        try:
-            event_sent_time = recent_samples.private_sndStamp[0].strftime(
-                self.time_format
+        print(f"*TRACE*dataframe:\n{dataframe}")
+        if dataframe.empty:
+            raise ValueError("Dataframe is empty")
+        # Get the sequence of summaryStates.
+        first = dataframe.summaryState[0]
```
[the remark this answers is not in the document]
Comment (rbovill) [reply QueryEfd.py:314]: Originally, I thought it made the code clearer and easier to understand.  Having tried the above suggestion, it seems to be pretty good, and there's less code.

File: QueryEfd.py (hunk 1 above)
Comment (rbovill) [reply QueryEfd.py]: Didn't realize I could just subtract the datetime objects; thought I had to convert to epoch first.  Keyword updated and simplified.

**Review by mareuter — APPROVED** [review]
Looks good to me. Thanks for making the changes.

**Pull request merged**